app/controllers/application_controller.rb

```diff
@@ -17,10 +17,9 @@ def visitor
   def redirect_after_login
     if session[:admin] == true
       redirect_to admin_path
-    elsif session[:return_to]
-      redirect_to session[:return_to]
     else
-      redirect_to new_order_path
+      redirect_to session[:return_to]
+      session[:return_to] = nil
     end
```

Comment:
Now a redirect_to is always set in the session. It'll only be called if there user is not an admin, and a forward_to has not been set which overrides the return_to.